### PR TITLE
Revert rust-lang/rust#46833 on stable

### PR DESCRIPTION
**Pull request by nikomatsakis**
This is a minimal diff to disable rust-lang/rust#46833 on stable (due to rust-lang/rust#48251). I have a more expansive diff for master that allows us to opt back in.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/test/run-pass/abort-on-c-abi.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // we never unwind through them.
 
 // ignore-emscripten no processes
+// ignore FIXME rust-lang/rust#48251 -- temporarily disabled
```
Comment (Mark-Simulacrum) [inline src/test/run-pass/abort-on-c-abi.rs]: This syntax appears to not be supported; we have to use `// ignore-test` instead I *think*...

**Review by Mark-Simulacrum — DISMISSED** [review]
Could you make the same change for the beta PR?

**Comment by nikomatsakis** [comment]
@Mark-Simulacrum done

File: src/librustc_mir/build/mod.rs [hunk 2]
```diff
@@ -372,7 +372,9 @@ fn should_abort_on_panic<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
     // unwind anyway. Don't stop them.
     if tcx.has_attr(tcx.hir.local_def_id(fn_id), "unwind") { return false; }
 
-    return true;
+    // FIXME(rust-lang/rust#48251) -- Had to disable abort-on-panic
+    // for backwardsa compatibility reasons.
```
Comment (carols10cents) [inline src/librustc_mir/build/mod.rs]: typo, not a big deal :)

> backwardsa

**Comment by nikomatsakis** [comment]
@rfcbot fcp merge

I propose that we merge this and issue a new point release -- this decision also applies to the [beta PR](https://github.com/rust-lang/rust/pull/48379) and [master PR](https://github.com/rust-lang/rust/pull/48380).

File: src/librustc_mir/build/mod.rs (hunk 2 above)
Comment (nikomatsakis) [reply src/librustc_mir/build/mod.rs]: are you KIDDING? That is the biggest of deals.

**Comment by ashleygwilliams** [comment]
@rfcbot reviewed

**Pull request closed without merging**